src/sqlfluff/rules/L013.py

```diff
@@ -48,6 +48,14 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         if segment.all(sp.is_type("select_clause_element")) and not children.any(
             sp.is_type("alias_expression")
         ):
+            # Ignore if it's a function with an EMITS clause as EMITS is equivalent to AS
+            if (
+                children.select(sp.is_type("function"))
+                .children()
+                .select(sp.is_type("emits_grammar"))
+            ):
+                return None
```

Comment:
@barrywhart my first rule change using your new rule functions! Is this the right way of using it?

Also trying to think of any queries that might not follow this. For example, if you wrapped the whole select clause in brackets:

```sql
(json_extract(json_str, '$.AFIELD', '$.BFIELD') emits (cola char(1), colb char(1)))
```

Then the above code would not catch it and L013 would flag but then not sure if that's actually OK and it SHOULD flag (I don't know Exasol so don't know how Emits works). Thoughts @sti0 ?

Maybe we should just search for ANY occurrence of `emits_grammar` in any children/grandchildren...etc. rather than just the direct children functions but not sure how easy that is (or even how accurate that would be and maybe above is actually right and rest SHOULD flag for L013)).

Reply:
> Maybe we should just search for ANY occurrence of `emits_grammar` in any children/grandchildren...etc. rather than just the direct children functions but not sure how easy that is (or even how accurate that would be and maybe above is actually right and rest SHOULD flag for L013)).

@barrywhart On this point, think we should add the `recursive_crawl` method to `Segments` to handle this sort of case? Not necessarily suggesting it has to be used in this rule, but would be useful in general.

Reply:
^pulling together a pr for recursive crawl

Reply:
Yeah I get how it works. So we can have:

```sql
select
    somecol,
    json_extract(json_str, '$.AFIELD', '$.BFIELD') emits (cola char(1), colb char(1))
from abc;
```

And that's basically equivalent to this:
```sql
select
    somecol,
    json_extract(json_str, '$.AFIELD') AS cola,
    json_extract(json_str, '$.BFIELD') AS colb
from abc;
```

The current fix caters for that.

However my question is, is below valid and equivalent?

```sql
select
    somecol,
    (json_extract(json_str, '$.AFIELD', '$.BFIELD') emits (cola char(1), colb char(1)))
from abc;
```

That is EMITS nested in another clause (brackets in this case)?

Because the current fix doesn't category that and ONLY works if the FUNCTION and EMITS is directly part of the SELECT clause.

But the more I think about it, the more it seems that brackets (or other nesting) aren't valid anyway, and EMITS probably DOES have to be part of the top level SELECT clause so think we're fine. And if it isn't we can always revisit later.

Reply:
@tunetheweb: Usage of the new rule functions looks 💯.

@jpy-git: +1 for updating the `Segments` class to provide `recursive_crawl()`. My intuition (based on a quick scan of the EXASOL docs) is that we wouldn't need to use that here, but I'm not certain. If we're in doubt, seems better to omit it for now. 🤷‍♂️

Reply:
@tunetheweb thanks for the fix. I will look into the `(json_extract(json_str, '$.AFIELD', '$.BFIELD') emits (cola char(1), colb char(1)))` thing tomorrow. 

Only for your interesst: `json_extract()` must use `emits()`. This 
```sql
select
    somecol,
    json_extract(json_str, '$.AFIELD') AS cola,
    json_extract(json_str, '$.BFIELD') AS colb
from abc;
```
must be written like this:
```sql
select
    somecol,
    json_value(json_str, '$.AFIELD') AS cola,
    json_value(json_str, '$.BFIELD') AS colb
from abc;
```